IVRv2/app/services/websocket_service.py

```
@@ -76,18 +77,45 @@ def cancel_bg_processes(self):
+            if self.is_connected and self._ws:
+                return
+
+            while not self.is_connected:
+                try:
+                    self._ws = await websockets.connect(self.connection_url)
+                    self.is_connected = True
+                    self.reconnect_attempts = 0
+                    logger.info(f"Connected to WebSocket: {self.connection_url}")
+                except Exception as e:
+                    self.reconnect_attempts += 1
+                    delay = min(1.0 * 2 ** self.reconnect_attempts, 30.0) + random.uniform(0, 1)
```

Ruff (0.15.13)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)